atomisticparsers/gromacs/parser.py

```diff
@@ -810,19 +810,66 @@ def get_gromacs_file(self, ext):
 
         return os.path.join(self._maindir, files[counts.index(min(counts))])
 
+    def get_all_gromacs_files(self, ext):
+        files = [d for d in self._gromacs_files if d.endswith(ext)]
+
+        if len(files) == 0:
+            return ''
+
+        all_files = {}
+        contain_basename, no_basename_match, counts = [], [], []
+        for f in files:
+            # We assume that the matching trajectory file has (or starts with)
+            # the same basename as the log file, e.g., out.log would correspond to
+            # out.tpr and out.trr and out.edr (or out_some.tpr ...).
+            # These files are therefore given first (second... ) priority in the dictionary.
+            if f.rsplit('.', 1)[0] == self._basename:
+                all_files[0] = [os.path.join(self._maindir, f)]
+            # ? `a in b` or `b.startswith(a) or b.endswith(a)`? How generic/specific do we want to be?
+            elif self._basename in f.rsplit('.', 1)[0]:
+                contain_basename.append(os.path.join(self._maindir, f))
```

Comment:
just directly append to all_files[1]

Reply:
unless you sort them as mentioned above, then add the sorting variable and leave contain_basename and sort below